analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/RuleChecks/InitializationVectorShouldBeRandomBase.cs

```diff
@@ -70,7 +70,7 @@ protected override ProgramState PreProcessSimple(SymbolicContext context)
         && property.Instance is { } propertyInstance
         && state.ResolveCaptureAndUnwrapConversion(propertyInstance).TrackedSymbol() is { } propertyInstanceSymbol
         && IsIVProperty(property, propertyInstanceSymbol)
-        && state[assignment.Value].HasConstraint(ByteCollectionConstraint.CryptographicallyWeak)
+        && state[assignment.Value]?.HasConstraint(ByteCollectionConstraint.CryptographicallyWeak) is { }
```

Comment:
I don't understand the line. `HasConstraint` returns `bool`, therefore the line seems to be equivalent to this:
```
&& state[assignment.Value] is { }
```
Didn't you intend to have `is true` instead? If yes, we're missing a test case that would break this because the constraint is not present where everything else matches.